integration/dev_test.go

```diff
@@ -295,7 +294,7 @@ func createModifiedKubeconfig(namespace string) ([]byte, string, error) {
 	}
 
 	contextName := "modified-context"
-	if isKind, _ := config.IsKindCluster(kubeConfig.CurrentContext); isKind {
+	if config.IsKindCluster(kubeConfig.CurrentContext) {
```

Comment:
I think this logic is just broken. 
Our kind cluster will be `modified-context` that leads to the error that can't load the image to `modified-context` cluster as it does not exist. Maybe we should rename the kind cluster to `modified-context`?

Reply:
I don't think that's what's happening:
+ Before we duplicate the kube context, we have a context that's usually `kind-kind` that references a cluster with the same name. To load images into kind, we have to get rid of the `kind-` prefix or suffix.
+ After we duplicate, the context is `kind-modified-context` but the cluster is still `kind-kind`. But the previous code never retrieved the name of the cluster. It assumed it was the same as the context. Which is the case by default but can be changed by the user.
+ What this code does is retrieve the actual cluster name, removed its prefix/suffix in order to get the *internal* name of the cluster

Reply:
L222 did exactly that prefix removal: `return true, strings.TrimPrefix(kubeContext, "kind-")` I thought you just refactored things around but did not change the logic - am I missing something?

Reply:
> After we duplicate, the context is `kind-modified-context` but the cluster is still `kind-kind`. But the previous code never retrieved the name of the cluster. It assumed it was the same as the context. Which is the case by default but can be changed by the user.

This code does not retrieve the name of the cluster either (in the test). 
The production code still uses the same prefix trimming logic - thus will still "think" the cluster is named `modified-context` instead of `kind` - unless

Reply:
You are missing the fact that the cluster name is not the context name.  Also, the kind internal cluster name is not the cluster name.

Here's an example of what can happen in the test:

Before we create a new context:
 + context name: `kind-kind`
 + cluster name: `kind-kind`
 + kind name: `kind`

After we create a new context:
 + context name: `kind-modified-context`
 + cluster name: `kind-kind` (This one doesn't change, that's where the bug comes from)
 + kind name: `kind`

This issue shows the case where users rename their context but keep the cluster name as is. We have to remove the prefix from the cluster name, not the context name.